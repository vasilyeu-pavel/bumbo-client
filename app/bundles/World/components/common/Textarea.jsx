import React, { PropTypes } from 'react';

const Textarea = props => (
  <div className='order-form-control'>
    <label className='order-form-control__label'>{props.label}</label>
    <textarea
      {...props}
      rows='3'
      className='order-form-control__input order-form-control__input--textarea'
    />
  </div>
);


Textarea.propTypes = {
  label: PropTypes.string,
};

export default Textarea;
