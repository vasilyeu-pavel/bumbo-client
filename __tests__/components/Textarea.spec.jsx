/* global expect, jest */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Textarea from '../../app/bundles/World/components/common/Textarea';

describe('<Textarea /> component', () => {
  const minProps = { label: 'description' };

  it('should render textarea element with label', () => {
    const textarea = renderer.create(<Textarea {...minProps} />).toJSON();

    expect(textarea).toMatchSnapshot();
  });

  it('should call the "onChange" method after textarea change', () => {
    const handler = jest.fn();
    const inputWrapper = shallow(<Textarea {...minProps} onChange={handler}/>);

    inputWrapper.find('textarea').simulate('change');

    expect(handler).toHaveBeenCalled();
  });
});
