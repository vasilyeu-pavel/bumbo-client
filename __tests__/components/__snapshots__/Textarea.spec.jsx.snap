exports[`<Textarea /> component should render textarea element with label 1`] = `
<div
  className="order-form-control">
  <label
    className="order-form-control__label">
    description
  </label>
  <textarea
    className="order-form-control__input order-form-control__input--textarea"
    label="description"
    rows="3" />
</div>
`;
